ovs/update/volumedriver/updater.py

```
@@ -224,54 +207,19 @@ def migrate_master_mds(cls, storagerouter, max_chain_size=100, group_timeout=10
-                    signature = MDSServiceController.ensure_safety.si(vdisk_guid, ensure_single_timeout=5)
-                    # Freeze freezes the task into its final form. This will net the async result object we'd normally get from delaying it
-                    tasks.append(signature.freeze())
-                    signatures.append(signature)
-                all_tasks.extend(tasks)
-                if signatures:
-                    cls.logger.info('Adding chain for VPool {} with tasks {}'.format(vpool_guid, ', '.join(t.id for t in tasks)))
-                    chains.append(chain(signatures))
-            # Add all chain signatures to a group for parallel execution
-            task_group = group(chains)
-            # Wait for the group result
-            async_result = task_group.apply_async()
-            cls.logger.info('Waiting for all tasks of group {}'.format(async_result.id))
-            try:
-                _ = async_result.get(timeout=group_timeout)
-            except TimeoutError:
-                cls.logger.warning('Migration took longer than expected. Revoking all non-started tasks')
-                revoked_tasks = []
-                for task in all_tasks:
-                    if task.state == 'PENDING':
-                        # Certain PENDING tasks cannot be revoked. It appears they're non-existent. Not even the workers know about them
-                        # @todo build a new result chain and wait for that
-                        task.revoke()
-                        revoked_tasks.append(task)
-                if revoked_tasks:
-                    cls.logger.warning('Revoked migration tasks: {}'.format(', '.join(t.id for t in revoked_tasks)))
-                cls.logger.warning('Waiting for the execution on the running migrations')
-                _ = async_result.get()
+                    safety_ensurer = SafetyEnsurer(vdisk_guid)
+                    safety_ensurer.ensure_safety()
```

You need to catch all exceptions here. You are calling the code inline which means the exception will be thrown up the stack of the callee as opposed to the stack of the executing worker

It's advisable to catch the EnsureSingleTimeOut and sleep in the meantime